### PR TITLE
framework/st_things: Fix issues to avoid crash

### PR DESCRIPTION
**Pull request by zhouxinhe**
1. strlen(NULL) cause crash.
2. in case of debugging, print THINGS_ERROR as a tag string cause crash.
@Hyunjun85

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cc58d3ea922ab85f107eecf339116a3f4c83f8bd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cc58d3ea922ab85f107eecf339116a3f4c83f8bd] - Code Rule Check (C++) OK.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 1]
```diff
@@ -924,25 +923,26 @@ int parse_things_cloud_json(const char *filename)
...
-		if (root != NULL) {
-			cJSON *cloud = cJSON_GetObjectItem(root, KEY_CLOUD);
-			if (cloud != NULL) {
-				cJSON *address = cJSON_GetObjectItem(cloud, KEY_CLOUD_ADDRESS);
-				if (address != NULL) {
-					memset(g_cloud_address, 0, (size_t) MAX_CLOUD_ADDRESS);
-					memcpy(g_cloud_address, address->valuestring, strlen(address->valuestring) + 1);
-					THINGS_LOG_D(TAG, "[CLOUD] CI Address : %s", g_cloud_address);
-					ret = 1;
+	} else {
```
Comment (jsdosa) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: Why did you add the else statement?
Each case already returns 0.
